### ISSUE TITLE
Cert and Sigh fails always

### ISSUE THREAD

**Issue by SunilProgramobil** [fastlane/fastlane#4251]
### New Issue Checklist

- [x ] Updated fastlane to the latest version
- [x ] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

##### Complete output when running fastlane, including the stack trace and command used

```
[15:29:31]: Starting login with user 'sunil.gandham@staples.com'
[15:29:35]: Successfully logged in
[15:29:35]: Fetching profiles...
[15:29:44]: [32mFound 1 matching profile(s)[0m
[15:29:44]: [33mDownloading provisioning profile...[0m
[15:29:45]: [32mSuccessfully downloaded provisioning profile...[0m
security: cert import failed: A default keychain could not be found.
security: problem decoding
[15:29:45]: [31mError parsing provisioning profile at path '/Users/Shared/Jenkins/Home/jobs/HubApp/workspace/InHouse_com.jenkins.test.ci.mobileprovision'[0m
[15:29:45]: [32m-----------------------------------[0m
[15:29:45]: [32m--- Step: clean_build_artifacts ---[0m
[15:29:45]: [32m-----------------------------------[0m
[15:29:45]: [32mCleaned up build artifacts 🐙[0m
[15:29:45]: [33m[33mVariable Dump:[0m[0m
[15:29:45]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta", :CERT_FILE_PATH=>"/Users/Shared/Jenkins/Home/jobs/HubApp/workspace/7D3BHGBNWM.cer", :CERT_CERTIFICATE_ID=>"7D3BHGBNWM"}
[15:29:45]: [31mundefined method `fetch' for nil:NilClass[0m

+------+-------------------------------------+-------------+
|                     [32mfastlane summary[0m                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | cocoapods                           | 2           |
| 4    | cert                                | 8           |
| 5    | sigh                                | 13          |
| 6    | clean_build_artifacts               | 0           |
+------+-------------------------------------+-------------+

[15:29:45]: [31mfastlane finished with errors[0m
[33m-------------------------------------------------------------------------------------------[0m
[33m😨  An error occured. Please enable crash reports using `fastlane enable_crash_reporting`.[0m
[33m👍  This makes resolving issues much easier and helps improve fastlane.[0m
[33m🔒  The reports will be stored securely on getsentry.com.[0m
[33m🙊  More information about privacy: https://github.com/fastlane/fastlane/releases/tag/1.33.3[0m
[33m-------------------------------------------------------------------------------------------[0m
/Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.41.2/lib/fastlane_core/provisioning_profile.rb:38:in `uuid': [31m[!] undefined method `fetch' for nil:NilClass[0m (NoMethodError)
	from /Library/Ruby/Gems/2.0.0/gems/sigh-1.6.1/lib/sigh/manager.rb:38:in `install_profile'
	from /Library/Ruby/Gems/2.0.0/gems/sigh-1.6.1/lib/sigh/manager.rb:25:in `start'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/actions/sigh.rb:20:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/runner.rb:143:in `block (2 levels) in execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/actions/actions_helper.rb:35:in `execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/runner.rb:129:in `block in execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/runner.rb:128:in `chdir'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/runner.rb:128:in `execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:34:in `block (2 levels) in parsing_binding'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/lane.rb:36:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/lane.rb:36:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/runner.rb:52:in `block in execute'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/runner.rb:46:in `chdir'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/runner.rb:46:in `execute'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/lane_manager.rb:46:in `cruise_lane'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/bin/fastlane:38:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.41.2/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/bin/fastlane:168:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/bin/fastlane:174:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:
```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.81.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    cocoapods
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    # match(type: "appstore") # more information: https://codesigning.guide
    cert
    sigh
    gym(scheme: "CI-Test")
    # sh "your_script.sh"
    # You can also use other beta testing services here (run `fastlane actions`)
  end
 
  lane :fail do
    puts ENV['FASTLANE_PASSWORD']
  end

  desc "Deploy a new version to the App Store"
  lane :appstore do
    # match(type: "appstore")
    # snapshot
    gym(scheme: "CI-Test")
    # frameit
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    if release_lane(lane)
      revert_version
    end
    clean
  end
  
  def clean
    # Don't clear provisioning profiles, as we store them in git
    clean_build_artifacts exclude_pattern: ".*\.mobileprovision"
  end

  def release_lane(lane)
    lane.to_s.include? 'release'
  end

  def revert_version
    Shenzhen::PlistBuddy.set plist, 'CFBundleShortVersionString', @previous_version
  end
  
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer

```

### Environment

fastlane version (run `fastlane -v`):fastlane 1.81.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)?yes

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)?

**Comment by vpolouchkine** [comment]
Can you inspect the provisioning profile path that is printed out for you and see what the file looks like and share the content?

**Comment by SunilProgramobil** [comment]
The file have is a proper provisioning profile and no issues with it. I will give you a setup walkthrough. I installed jenkins in my machine (macbook prob) as different user in shared and running it.I see this problem only when I run it through jenkins if I run fastlane command in normal user it works fine. If that helps you.

**Comment by vpolouchkine** [comment]
Perhaps you can leverage the `setup_jenkins` action, see https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Jenkins.md#setup_jenkins

**Comment by SunilProgramobil** [comment]
Thanks for the suggestion will try and see. Right now I am running into weird issue with Cocoapods.

 Variable Dump:
[19:37:40]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta"}
[19:37:40]: Exit status of command 'pod install' was 1 instead of 0.

――― MARKDOWN TEMPLATE ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

### Command

```
/usr/local/bin/pod install
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?


### Stack

```
   CocoaPods : 0.39.0
        Ruby : ruby 2.0.0p645 (2015-04-13 revision 50299) [universal.x86_64-darwin15]
    RubyGems : 2.0.14
        Host : Mac OS X 10.11.2 (15C50)
       Xcode : 7.3 (7D175)
         Git : git version 2.6.4 (Apple Git-63)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ ee5e9bfb468327cddc97538366d8d156fb228970
```

### Plugins

```
cocoapods-plugins : 0.4.2
cocoapods-search  : 0.1.0
cocoapods-stats   : 0.6.2
cocoapods-trunk   : 0.6.4
cocoapods-try     : 0.5.1
```

### Podfile

```ruby
# Pods setup
platform :ios, '8.0'
inhibit_all_warnings!
use_frameworks!
source 'https://github.com/CocoaPods/Specs.git'
link_with ['Complaints']


# 3rdParty pods
pod 'MMDrawerController', '~> 0.5.7'
pod 'FBSDKLoginKit', '~> 4.9'
pod 'AFDateHelper', '~> 3.1'
pod 'MBProgressHUD', '~> 0.9'
pod 'TextImageButton'
pod 'SZTextView'
pod 'INTULocationManager', '~> 4.1'
pod 'IQKeyboardManagerSwift', '~> 4.0'
pod 'ReachabilitySwift', '~> 2.3'
pod 'GoogleMaps'
pod 'GRKGradientView', '~> 1.0'
pod 'Alamofire', '~> 3.0'
pod 'AlamofireImage', '~> 2.4'
pod 'DKImagePickerController', '~> 3.1'
pod 'TOCropViewController', '~> 1.3'
pod 'Locksmith'
pod 'MagicalRecord', '~> 2.3'
```

### Error

```
Errno::EACCES - Permission denied - /Users/Shared/Jenkins/Home/jobs/Complaints/workspace/Pods/Local Podspecs
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:245:in `mkdir'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:245:in `fu_mkdir'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:219:in `block (2 levels) in mkdir_p'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:217:in `reverse_each'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:217:in `block in mkdir_p'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:203:in `each'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:203:in `mkdir_p'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.39.0/lib/cocoapods/sandbox.rb:106:in `prepare'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:118:in `block in prepare'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.39.0/lib/cocoapods/user_interface.rb:140:in `message'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:117:in `prepare'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.39.0/lib/cocoapods/installer.rb:104:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.39.0/lib/cocoapods/command/project.rb:71:in `run_install_with_update'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.39.0/lib/cocoapods/command/project.rb:101:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-0.9.1/lib/claide/command.rb:312:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.39.0/lib/cocoapods/command.rb:47:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.39.0/bin/pod:44:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

――― TEMPLATE END ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

[!] Oh no, an error occurred.

Search for existing GitHub issues similar to yours:
https://github.com/CocoaPods/CocoaPods/search?q=Permission+denied+-+%2FUsers%2FShared%2FJenkins%2FHome%2Fjobs%2FComplaints%2Fworkspace%2FPods%2FLocal+Podspecs&type=Issues

If none exists, create a ticket, with the template displayed above, on:
https://github.com/CocoaPods/CocoaPods/issues/new

Be sure to first read the contributing guide for details on how to properly submit a ticket:
https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md

Don't forget to anonymize any private data!


-------------------------------------------------------------------------------------------
😨  An error occured. Please enable crash reports using `fastlane enable_crash_reporting`.
👍  This makes resolving issues much easier and helps improve fastlane.
🔒  The reports will be stored securely on getsentry.com.
🙊  More information about privacy: https://github.com/fastlane/fastlane/releases/tag/1.33.3
-------------------------------------------------------------------------------------------
/Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/documentation/docs_generator.rb:39:in `write': [!] Permission denied - ./fastlane/README.md (Errno::EACCES)
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/documentation/docs_generator.rb:39:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/lane_manager.rb:55:in `cruise_lane'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/bin/fastlane:38:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.41.2/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/bin/fastlane:168:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.81.0/bin/fastlane:174:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'

**Comment by SunilProgramobil** [comment]
That happens when I run fastlane command from jenkins user

**Comment by asfalcone** [comment]
@SunilProgramobil Can you please open an issue with cocoapods since that error is coming from the cocoapods gem? Seems like a permissions problem (Errno::EACCES - Permission denied). Maybe you need to give that user more permissions?

Their error output includes the following:

Search for existing GitHub issues similar to yours:
https://github.com/CocoaPods/CocoaPods/search?q=Permission+denied+-+%2FUsers%2FShared%2FJenkins%2FHome%2Fjobs%2FComplaints%2Fworkspace%2FPods%2FLocal+Podspecs&type=Issues

If none exists, create a ticket, with the template displayed above, on:
https://github.com/CocoaPods/CocoaPods/issues/new

**Comment by SunilProgramobil** [comment]
@asfalcone That last issue which i mentioned with cocoapods makes sense and will take care of it coming back to the original issues raised. @vpolouchkine  mentioned to use setup_jenkins to get it resolved but it didnt work.  I get this always Error parsing provisioning profile

**Comment by SunilProgramobil** [comment]
I can run fastlane on normal user when switch to jenkins user I am facing and issue that code sign error. @asfalcone @vpolouchkine

**Comment by SunilProgramobil** [comment]
The only problem I am facing now is accessing certificates and provisioning profiles using fastlane from Jenkins User.
Here is some insight on how Jenkins is setup.

XXXX is the admin user of the mac and I installed Jenkins in /Users/Shared/ 
if that helps

**Comment by SunilProgramobil** [comment]
[31m❌  Code Sign error: No provisioning profiles found: No non–expired provisioning profiles were found.[0m

**Comment by TKBurner** [comment]
Hi @SunilProgramobil Thanks for following up. Can you run cert --verbose and share the output? We really want to look into this!

Thanks! 👍

**Comment by SunilProgramobil** [comment]
@TKBurner  I was closely working on it and after so much digging into the folder structure of Jenkins User I found that there is no folder MobileDevice/Provisioning Profiles in Library of Jenkins User. 
I am guessing fastlane is trying to find profiles from or installing into this folder. So I had to manually create those folders in Jenkins User and then run sigh command in the following path /User/Jenkins/Library/MobileDevice/Provisioning Profiles  could install profiles succesfully and app could use it later. 
However there still exists a problem if I run the same commands (having them within the fastfile) and run it in execute shell of Jenkins.
I am not sure if I got a solution for this. I hope this helps
